MediaBrowser.Controller/Entities/Folder.cs

```diff
@@ -842,26 +842,27 @@ private bool RequiresPostFiltering(InternalItemsQuery query)
 
         private static BaseItem[] SortItemsByRequest(InternalItemsQuery query, IReadOnlyList<BaseItem> items)
         {
-            var ids = query.ItemIds;
             int size = items.Count;
-
             // ids can potentially contain non-unique guids, but query result cannot,
             // so we include only first occurrence of each guid
             var positions = new Dictionary<Guid, int>(size);
             int index = 0;
-            for (int i = 0; i < ids.Length; i++)
+            foreach (var id in query.ItemIds)
             {
-                if (positions.TryAdd(ids[i], index))
+                if (positions.TryAdd(id, index))
                 {
                     index++;
                 }
             }
 
             var newItems = new BaseItem[size];
-            for (int i = 0; i < size; i++)
+            foreach (var item in items)
             {
-                var item = items[i];
-                newItems[positions[item.Id]] = item;
+                int x = positions[item.Id];
+                if (x != -1)
```

Comment:
How can index ever be -1 when it starts at 0 and is incremented?

Isn't the problem simply that `size` is too large if there are duplicates? Replace size with `.Count`.

Reply:
size = .Count

Re-looking at the code the only way it could error is if the number of unique items in Query.ItemIds > items.count

Reply:
size should be `positions.Count`

Reply:
> Re-looking at the code the only way it could error is if the number of unique items in Query.ItemIds > items.count

Looks that way. But the reason for ItemIds being bigger is that one or more of those ids no longer exist in the database. Even if you fix the IndexOutOfRange error, you'll run into another problem of one of the indices not being populated in `newItems`, which I'm thinking will cause a crash when serializing it.